### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by JJaug**

### CODE REVIEW

File: AnagramSolver.BusinessLogic/Classes/AnagramSolver.cs [hunk 1]
```diff
@@ -8,38 +8,46 @@ namespace AnagramSolver.BusinessLogic.Classes
...
+            command = command.Replace(" ", string.Empty);
 
-            foreach (var word in allWords)
+            var commandChars = command.ToList();
+            var remainingChars = commandChars;
+            var result = string.Empty;
+
+            sentenceMaker(remainingChars);
+            
+            void sentenceMaker(List<char> remainingChars)
```
Comment (KristupasLunskas) [inline AnagramSolver.BusinessLogic/Classes/AnagramSolver.cs:28]: Naming convention. Methods should start with capital letter (https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions)

Nested methods has it's drawbacks. You should experience them when writing tests for this class.

File: AnagramSolver.Cli/Program.cs [hunk 2]
```diff
@@ -9,45 +9,46 @@ class Program
...
-            var config = builder.Build();
-            var minLength = config["MinWordLength"];
-            int minLengthAsNumber = Int32.Parse(minLength);
-            var maxAnagrams = config["MaxNumberOfAnagrams"];
-            int maxAnagramsAsNumber = Int32.Parse(maxAnagrams);
-            Console.WriteLine(maxAnagramsAsNumber);
-
+            int minLength, maxAnagrams;
+            string filePath;
+            ConfigureAppSettings(out minLength, out maxAnagrams, out filePath);
```
Comment (KristupasLunskas) [inline AnagramSolver.Cli/Program.cs:14]: Out is not a bad practice when used correctly. Here we can see that if we apply more configuration, we have to expand with new parameter and make a variable for it. There are other ways of structuring multiple fields together  (wink wink) and using methods return value.

File: AnagramSolver.Cli/Program.cs [hunk 3]
```diff
@@ -9,45 +9,46 @@ class Program
...
                     Console.WriteLine($"Zodis turi buti bent is {minLength} raidziu");
                     command = Console.ReadLine();
                 }
-                if(command == "exit")
+                if (command == "exit")
                 {
                     break;
                 }
-                var listOfAnagrams = result.GetAnagrams(command);
+                var listOfAnagrams = result.GetAnagrams(command, minLength, filePath);
```
Comment (KristupasLunskas) [inline AnagramSolver.Cli/Program.cs:32]: This is fine for now, however if we apply more configuration in appsettings.json we have to append new parameter for each line. A method with more than three parameters are questionable.

`Bob Martin says the ideal number of parameters is 0 and that 3 is stretching it.`
(https://www.informit.com/articles/article.aspx?p=1375308)

Strongly recommend to check for better practices (later).
(https://stackoverflow.com/questions/48948905/how-to-access-appsettings-from-another-project/48951371)

Comment (JJaug) [reply AnagramSolver.Cli/Program.cs:32]: okay, will do

File: AnagramSolver.Cli/Program.cs (hunk 2 above)
Comment (JJaug) [reply AnagramSolver.Cli/Program.cs:14]: okay

**Pull request merged**